pkg/core/transactor/transactor.go

```diff
@@ -0,0 +1,186 @@
+package transactor
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	ristretto "github.com/bwesterb/go-ristretto"
+	cfg "github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/block"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/database"
+	"github.com/dusk-network/dusk-blockchain/pkg/core/database/heavy"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet"
+	"github.com/dusk-network/dusk-blockchain/pkg/wallet/transactions"
+	"github.com/dusk-network/dusk-wallet/key"
+)
+
+// TODO: rename
+type Transactor struct {
+	w  *wallet.Wallet
+	db database.DB
+}
+
+// Instantiate a new Transactor struct.
+func New(w *wallet.Wallet, db database.DB) *Transactor {
+	if db == nil {
+		_, db = heavy.CreateDBConnection()
+	}
+
+	return &Transactor{
+		w:  w,
+		db: db,
+	}
+}
+
+func (t *Transactor) CreateStandardTx(amount uint64, address string) (transactions.Transaction, error) {
+	if err := t.syncWallet(); err != nil {
+		return nil, err
+	}
+
+	// Create a new standard tx
+	// TODO: customizable fee
+	tx, err := t.w.NewStandardTx(cfg.MinFee)
+	if err != nil {
+		return nil, err
+	}
+
+	// Turn amount into a scalar
+	amountScalar := ristretto.Scalar{}
+	amountScalar.SetBigInt(big.NewInt(0).SetUint64(amount))
+
+	// Send amount to address
+	tx.AddOutput(key.PublicAddress(address), amountScalar)
+
+	// Sign tx
+	err = t.w.Sign(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx, nil
+}
+
+func (t *Transactor) CreateStakeTx(amount, lockTime uint64) (transactions.Transaction, error) {
+	if err := t.syncWallet(); err != nil {
+		return nil, err
+	}
+
+	// Turn amount into a scalar
+	amountScalar := ristretto.Scalar{}
+	amountScalar.SetBigInt(big.NewInt(0).SetUint64(amount))
+
+	// Create a new stake tx
+	tx, err := t.w.NewStakeTx(cfg.MinFee, lockTime, amountScalar)
+	if err != nil {
+		return nil, err
+	}
+
+	// Sign tx
+	err = t.w.Sign(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx, nil
+}
+
+func (t *Transactor) CreateBidTx(amount, lockTime uint64) (transactions.Transaction, error) {
```

Comment:
The suggestion here was more about making Transactor concurrent-safe.

Reply:
Ah I see. Apart from the walletDB though, there's not much that should be protected on this struct. However, after a quick look I believe it would be a good idea to lock while syncing, so I will implement it.

Reply:
Up to you.